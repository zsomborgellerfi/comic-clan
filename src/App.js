import React from "react";
import Header from "./components/Header";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Redirect,
} from "react-router-dom";
import Comics from "./containers/Comics";
import ComicDetails from "./containers/ComicDetails";
import { comicFilterKeys } from "./utils/comicFilterKeys";
import ScrollToTop from "./components/ScrollToTop";
import "./styles/_styles.scss";

const App = () => {
  return (
    <Router>
      <Header />
      <ScrollToTop />
      <Switch>
        <Route exact path="/comics/:category?" component={Comics} />
        <Route exact path="/comic/:name" component={ComicDetails} />
        <Redirect to={`/comics/${comicFilterKeys[0]}`} />
      </Switch>
    </Router>
  );
};

export default App;
