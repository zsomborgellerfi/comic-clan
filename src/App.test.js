import React from "react";
import { shallow } from "enzyme";
import App from "./App";

describe("App", () => {
  const wrapper = shallow(<App />);

  it("should render App correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 2 Routes correctly", () => {
    expect(wrapper.find("Route")).toHaveLength(2);
  });
});
