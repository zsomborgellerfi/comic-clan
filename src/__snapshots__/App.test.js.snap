// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render App correctly 1`] = `
<BrowserRouter>
  <Header />
  <withRouter(ScrollToTop) />
  <Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Comics)",
          "type": [Function],
        }
      }
      exact={true}
      path="/comics/:category?"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(ComicDetails)",
          "type": [Function],
        }
      }
      exact={true}
      path="/comic/:name"
    />
    <Redirect
      to="/comics/year"
    />
  </Switch>
</BrowserRouter>
`;
